File: leetcode-python/Quests/Array_I.py
class Solution:

    #Array I : Q1. Concatenation of Array
    def getConcatenation(self, nums: list[int]) -> list[int]:
        lenght: int = len(nums)
        output: list[int] = [0] * lenght * 2

        for i in range(0, lenght):
            output[i] = nums[i]
            output[lenght + i] = nums[i]

        return output

    #Array I : Q2. Shuffle the Array
    def shuffle(self, nums: list[int], n: int) -> list[int]:
        output: list[int] = [0] * n * 2

        for i in range(n):
            output[i * 2] = nums[i]
            output[i * 2 + 1] = nums[i + n]

        return output

    #Array I : Q3. Max Consecutive Ones
    def findMaxConsecutiveOnes(self, nums: list[int]) -> int:

        max_occurence: int = 0
        current_occurence: int = 0

        for num in nums:

            if num == 1:
                current_occurence += 1
            else:

                if current_occurence > max_occurence:
                    max_occurence = current_occurence

                current_occurence = 0

        return max(max_occurence, current_occurence)

File: leetcode-java/src/quest/ArrayI.java
package quest;

public class ArrayI
{
    // Array I : Q1. Concatenation of Array
    public int[] getConcatenation(int[] nums)
    {
        int length = nums.length;
        int[] output = new int[length * 2];

        for (int i = 0; i < length; i++)
        {
            output[i] = nums[i];
            output[i + length] = nums[i];
        }

        return output;
    }

    // Array I : Q2. Shuffle the Array
    public int[] shuffle(int[] nums, int n) {

        int[] result = new int[2 * n];

        for (int i = 0; i < n; i++) {

            result[i * 2] = nums[i];
            result[i * 2 + 1] = nums[i + n];
        }

        return result;
    }

    // Array I : Q3. Max Consecutive Ones
    public int findMaxConsecutiveOnes(int[] nums) {

        int result = 0;
        int currentCount = 0;

        for (int number : nums) {

            if (number == 1) {

                currentCount++;

                if (currentCount > result) {
                    result = currentCount;
                }
            }
            else {
                currentCount = 0;
            }
        }

        return result;
    }
}
